### PR TITLE
Linguist fixes based on raiph's PR

### PR DESCRIPTION
**Pull request by 2colours**
I modified the Raku sources (also did some unquoting the escape the... escape hell), then generated the grammars using generate-grammars.raku (get used to it, folks).

The changes in the JSON file should be the same as for https://github.com/Raku/atom-language/pull/102.

### CODE REVIEW

**Comment by raiph** [comment]
I'm currently having serious problems getting reliable / significant internet access. I have managed to get a few minutes now to try catch things like this message but I've only got 3 minutes right now so only time to say yes, I want to react to what you've written, and plan to write a comment tonight, but won't get a chance to post it until tomorrow at the earliest and perhaps several days at the hope for worst.

**Comment by raiph** [comment]
I'm having a tough time contributing anything online right now for a wide range of reasons and even this seemingly simple task is beyond me. (I can barely read the screen as I type this,  with the best I can manage being a screen area an inch or so deep and and few wide so I can make sure words in this comment are spelled right being about as good as I can get things! (Don't ask. Maybe in a month when I'm past all the issues that are currently piling on. OK, to at least explain the screen thing I'm outdoors where I can get slow/unreliable wifi but the sun is shining and in the shade the wifi connection breaks...)

Suffice to say, I can't check this merge for three main reasons. 1, I can't reliably read the characters on the screen. 2) I can't relax enough to focus. 3) Even if I could fix 1 and 2 I can't test any of it.

If it's not a big deal if merging it breaks things, because you can just back the changes out and there'll not be any significant consequence of that, then merging sounds plausibly reasonable. Otherwise, I hope to be living in my new home, and have a proper internet connection, and be able to sort out 1 and 2 above (but still not 3) within at most a month (and hopefully sooner).

**Comment by 2colours** [comment]
Thank you for the follow-up!

No worries. To be honest, I haven't made up my mind if I want to rush forward recklessly, do further investigation about getting a working test environment again, or put it aside until you can take a more elaborate look on things... either way, your contribution is welcome and I'm happy for anybody getting involved in any way to any extent. ^^

File: grammars/raku.tmLanguage.json [hunk 1]
```diff
@@ -2779,7 +2779,7 @@
               "name": "variable.other.identifier.raku"
             }
           },
-          "match": "(?x)\n(\\$|@|%|&)\n(\\.|\\*|:|!|\\^|~|=|\\?)?\n(\n    (?:[\\p{Alpha}_])           # Must start with Alpha or underscore\n    (?:\n       [\\p{Digit}\\p{Alpha}_]  # have alphanum/underscore, or a ' or -\n    |                           # followed by an Alpha or underscore\n       [\\-'] [\\p{Alpha}_]\n    )*\n)",
+          "match": "(?x)\n(\\$|@|%|&)\n(\\.|\\*|:|!|\\^|~|=|\\?)?\n(\n    (?:[\\pL\\pM_])           # Must start with Alpha or underscore\n    (?:\n       [\\\\p{Digit}\\pL\\pM_]  # have alphanum/underscore, or a ' or -\n    |                           # followed by an Alpha or underscore\n       [\\-'] [\\pL\\pM_]\n    )*\n)",
```
Comment (raiph) [inline grammars/raku.tmLanguage.json]: Is `\\\\p{Digit}` a mistake?

Comment (2colours) [reply grammars/raku.tmLanguage.json]: Yes, I think that was a mistake but it had been fixed by the time you saw the PR, no? Github even says that the file you commented on is an outdated view.

Comment (raiph) [reply grammars/raku.tmLanguage.json]: > it had been fixed by the time you saw the PR, no?

I don't know. I just got an opportunity to do stuff online, pulled up this PR last night, did a review, and it was the only thing I spotted that didn't look right.

I didn't notice anything about anything being outdated. I do see an "outdated" tag now though. Maybe I missed it? I've only tried to use the PR review feature a few times before, and I don't think I've reviewed an outdated file before. (I'm a bit surprised they don't disallow reviewing of an outdated file, or at least make the warning more prominent. Then again, now I've made that mistake I will be looking out for signs of it in the future, so maybe it doesn't need to be more prominent.)

Anyhow, suffice to say I've now reviewed this PR and it "looks good" to me. (That said, I've not tested it, and won't be testing it.)

Comment (raiph) [reply grammars/raku.tmLanguage.json]: Weirdly, after writing my prior comment, and then clicking "View reviewed changes", I arrived at a new page (this one) that shows my review comment above, and your reply, and then the comment I just wrote, but does *not* show the "outdated" tag against the file name.

And I'm now no longer in review mode (there's a green button below this comment that says "Start a review") and am just about to click the grey "Add single comment" button that's next to the green "Start a review" button. I've no idea if this comment will land underneath the ones above that I just mentioned but let's see...

**Comment by 2colours** [comment]
@raiph so do you think it is okay if I merge this now? Don't worry about the tests; eventually, there will be a new way for testing but right now it doesn't matter that much.

**Comment by raiph** [comment]
I'd not bet money it's OK because I haven't tested it, but then it's low stakes right? If it's wrong, someone says something, you back it out, no harm done, right? If so, feel free to merge it.

**Pull request merged**